lib/seadsa/ShadowMem.cc

```diff
@@ -376,8 +376,8 @@ class ShadowMemImpl : public InstVisitor<ShadowMemImpl> {
     if (it != offsetMap.end()) return it->second;
 
     // -- not found, allocate new shadow variable
-    AllocaInst *inst = new AllocaInst(m_Int32Ty, 0U /* Address Space */, "",
-(Instruction*)nullptr);
+    AllocaInst *inst = new AllocaInst(m_Int32Ty, 0U /* Address Space */,
```

Comment:
this does not look right. The signature seems to be for allocating arrays, but only a single `i32` is allocated.

Reply:
The correct constructor to use is this one: https://llvm.org/doxygen/Instructions_8cpp_source.html#l01322

Reply:
since a name is required, some name can be given

Reply:
This change is somehow outdated. Could you take a look at the latest version? Btw, I tried to the constructor you mentioned but got segfaults. That's why I switched to another constructor.

Reply:
I did. Please squash commit history and resubmit. Then I can comment again.

You seem to be using a constructor that you do not understand, while a simpler constructor causes a crash. This indicates some misunderstanding. If a change that should be ok causes a crash, it is an indication that something else maybe wrong somewhere. We need to get to the bottom of why the crash is happening and whether the solution you are proposing is correct. I am particularly worried about empty `Align` attribute since I don't know whether it means some hard-coded alignment or standard alignment for the architecture.